### PR TITLE
Adding autoflake to the project

### PR DESCRIPTION
**Pull request by ShmuelTreiger**
Autoflake removes unused imports

Pros:
- Removes tons of stuff that's currently been left behind
- Nifty

Cons:
- `autoflake -r -i --remove-all-unused-imports . && isort . && black .` is messy
- autoflake seems to be a bit slow
- necessitates `#noqa` on rather long imports, in part because:
- black interacts with it weirdly. I tried moving imports which need `#noqa` on their own line (i.e. having two `from X import *`), but black recombined most of them (but weirdly not all of them, so I recombined the rest manually)

Neutral:
- isort and black seem unaffected by `#noqa`, so no need to worry about that interaction

Alternative option:
Instead of adding autoflake to our "process", we could just run it ~once a month

**Backend checklist**
- [x] Formatted my code by running `isort . && black .` in `app/backend`
- [x] All tests pass
- [ ] Run the backend locally and it works

### CODE REVIEW

**Comment by kalvdans** [comment]
> `autoflake -r -i --remove-all-unused-imports . && isort . && black .` is messy

True, autoflake has no config file, which is reported as https://github.com/myint/autoflake/issues/59

**Comment by kalvdans** [comment]
Nice cleanup! I like.

> Instead of adding autoflake to our "process", we could just run it ~once a month

I'm 50/50 on this. I would say we merge this PR and try it out, if it is a pain we can always go back to running it regularly.

**Review by kalvdans — DISMISSED** [review]
I approve but let's also hear other developers opinions.

**Comment by aapeliv** [comment]
Not working for me:

```
~/git/couchers/app/backend$ autoflake -r -i --remove-all-unused-imports . && isort . && black .
Traceback (most recent call last):
  File "/Users/aapeli/git/couchers/app/backend/venv/bin/isort", line 5, in <module>
    from isort.main import main
  File "/Users/aapeli/git/couchers/app/backend/venv/lib/python3.9/site-packages/isort/main.py", line 13, in <module>
    from . import __version__, api, files, sections
  File "/Users/aapeli/git/couchers/app/backend/venv/lib/python3.9/site-packages/isort/api.py", line 10, in <module>
    from isort import core
  File "/Users/aapeli/git/couchers/app/backend/venv/lib/python3.9/site-packages/isort/core.py", line 6, in <module>
    import isort.literal
  File "/Users/aapeli/git/couchers/app/backend/venv/lib/python3.9/site-packages/isort/literal.py", line 10, in <module>
    from isort.settings import DEFAULT_CONFIG, Config
  File "/Users/aapeli/git/couchers/app/backend/venv/lib/python3.9/site-packages/isort/settings.py", line 262, in <module>
    _DEFAULT_SETTINGS = {**vars(_Config()), "source": "defaults"}
  File "<string>", line 97, in __init__
  File "/Users/aapeli/git/couchers/app/backend/venv/lib/python3.9/site-packages/isort/settings.py", line 230, in __post_init__
    raise ValueError(
ValueError: The python version 3 is not supported. You can set a python version with the -py or --python-version flag. The following versions are supported: ()
```

**Comment by aapeliv** [comment]
It's taking a **really** long time for me... is it possibly trying to process everything in `venv/`?

**Review by aapeliv — CHANGES_REQUESTED** [review]
Yeah, it's not working for me. See the error above. It takes forever to run then fails with that error message and somehow screws up my environment in the process

**Comment by kalvdans** [comment]
With warm filesystem caches:
```
(venv) chn@brahe:~/couchers/couchers/app/backend$ time autoflake  -r -i --remove-all-unused-imports .

real    0m3,758s
user    0m3,531s
sys     0m0,051s
(venv) chn@brahe:~/couchers/couchers/app/backend$ time black .
All done! ✨ 🍰 ✨
112 files left unchanged.

real    0m1,849s
user    0m7,211s
sys     0m0,144s
(venv) chn@brahe:~/couchers/couchers/app/backend$ time isort .
Skipped 1 files

real    0m1,057s
user    0m0,877s
sys     0m0,210s
```

**Comment by kalvdans** [comment]
> Yeah, it's not working for me. See the error above. It takes forever to run then fails with that error message and somehow screws up my environment in the process

Don't create your venv inside the git repo :)

**Comment by aapeliv** [comment]
> > Yeah, it's not working for me. See the error above. It takes forever to run then fails with that error message and somehow screws up my environment in the process
> 
> Don't create your venv inside the git repo :)

I always do! The canonical placement of the virtualenv is as a sibling to `requirements.txt`!

**Comment by kalvdans** [comment]
> I always do! The canonical placement of the virtualenv is as a sibling to `requirements.txt`!

Makes some sense since we have different requirements for media and backend, but I add them both to the same venv. Run the tools in the src directory then :)

**Comment by ShmuelTreiger** [comment]
@aapeliv try adding '--exclude venv' before the period

Like @kalvdans I have one environment for the whole couchers project, though, tbf, I only work on the backend

**Comment by kalvdans** [comment]
> @aapeliv try adding '--exclude venv' before the period

Should we replace `.` with `src` in this Pull Request and call it a day?

**Comment by ShmuelTreiger** [comment]
That works for me, I was just trying to isolate aapeli's problem before solving it

**Review by kalvdans — DISMISSED** [review]
Looks good to me still. By the way, what is the `[skip CI]` markup doing in your commit comment?

**Comment by ShmuelTreiger** [comment]
Skip continuous integration

So the tests don't run (I only changed documentation)

**Comment by kalvdans** [comment]
> So the tests don't run (I only changed documentation)

Oh, I didn't know about that, I learn something new every day :)

File: docs/database.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ cd backend
 DATABASE_CONNECTION_STRING="postgresql://postgres:203d805f4b62c0a1b2f1f6b82d4583dfe563ec1619b83ce22ee414e8376a25e7@localhost:6545/postgres" PYTHONPATH=src alembic revision --autogenerate -m "Modify the database"
 
 # now format the migration
-isort . && black .
+autoflake -r -i --remove-all-unused-imports . && isort . && black .
```
Comment (aapeliv) [inline docs/database.md]: `.` -> `src`?

**Review by aapeliv — DISMISSED** [review]
Yeah, I'm happy for this to be merged! Thanks for doing this!

File: docs/database.md (hunk 1 above)
Comment (aapeliv) [reply docs/database.md]: This should be `autoflake -r -i --remove-all-unused-imports src`, right?

**Review by aapeliv — CHANGES_REQUESTED** [review]

File: docs/database.md (hunk 1 above)
Comment (ShmuelTreiger) [reply docs/database.md]: Yup. My B. Thought I'd done it

**Review by aapeliv — APPROVED** [review]
LGTM. Happy to merge when CI goes green!

**Pull request merged**